ethereum/ethevents/events.go

```diff
@@ -290,13 +291,37 @@ func (ev *EthereumEvents) SubscribeEthereumEventLogs(ctx context.Context, fromBl
 		go ev.runEventProcessor(ctx)
 	}
 
+	// check eth connection
+	connFailure := make(chan bool, 1)
+	go func(chan bool) {
+		for {
+			time.Sleep(time.Second * 5)
+			tctx, cancel := context.WithTimeout(ctx, types.EthereumReadTimeout)
+			defer cancel()
+			block, err := ev.VotingHandle.EthereumClient.BlockByNumber(tctx, nil)
+			if err != nil {
+				log.Warnf("cannot check ethereum connection while running the event processor, %s", err)
+				connFailure <- false
+				return
+			}
+			ev.EthereumLastKnownBlock = block.Number().Int64()
```

Comment:
This var is never used and also it looks like racy, I think sync.Atomic should be used here

Reply:
yep, so finally adding the commit I mentioned regarding this case here.